src/NATS.Client.JetStream/NatsJSConsumer.cs

```diff
@@ -0,0 +1,199 @@
+using System.Runtime.CompilerServices;
+using NATS.Client.Core;
+using NATS.Client.JetStream.Models;
+
+namespace NATS.Client.JetStream;
+
+public class NatsJSConsumer
+{
+    private readonly NatsJSContext _context;
+    private readonly ConsumerInfo _info;
+    private readonly ConsumerOpts _opts;
+    private readonly string _stream;
+    private readonly string _consumer;
+
+    public NatsJSConsumer(NatsJSContext context, ConsumerInfo info, ConsumerOpts opts)
+    {
+        _context = context;
+        _info = info;
+        _opts = opts;
+        _stream = _info.StreamName;
+        _consumer = _info.Name;
+    }
+
+    public async IAsyncEnumerable<NatsJSMsg<T?>> ConsumeAsync<T>(int maxMsgs, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Comment:
Probably need a check here to make sure this is a Pull consumer, and if not throw an Exception explaining that this API only supports Pull consumers

Reply:
Added checks in consumer create method.

I believe pull/push consumer selection can only be done at creation. Are you saying users should be able to create pull consumers but not be able to use them? Which makes sense then to place a check here., but if that's the case I would say maybe we shouldn't support even the idea of pull/push consumers. I think consumers are complicated enough we can do with one less concept to reduce the cognitive overload. If we have to do it for KV for example we can support that internally maybe?